### ISSUE TITLE
Review-2018.04.15-杨磊

### ISSUE THREAD

**Issue by DIYer22** [DIYer22/weekly-review#112]
Weekly Report by Lei Yang -- April 15, 2018
### 二. 看了部分 [ An Introduction to Image Synthesis with Generative Adversarial Nets](https://arxiv.org/abs/1803.04469)
今年3月份(18.03)的发布的一篇用 GAN 做 Image Synthesis 的综述.    
总结了几种常见的 Image Synthesis GAN的方法

#### 结构上的改进

1. GAN with Auxiliary Classifier：   
![2018 04 15 c](https://user-images.githubusercontent.com/10448025/38772800-a784d08a-4071-11e8-8151-e8cfb0f3386c.png)    
Auxiliary Classifier 能提高生成图片的清晰度，并缓解 mode collapse

2. GAN with Encoder   
![2018 04 15 bigan](https://user-images.githubusercontent.com/10448025/38772802-ac4f1df0-4071-11e8-8008-3061e1c9c6fb.png)    
单独加了个Encoder `E`， 感觉这个Loss 还可以加点东西 比如 `G => min||E(G(z))-z||`

3. GAN with Variational Auto-Encoder   
![2018 04 15 vaegan](https://user-images.githubusercontent.com/10448025/38772803-b03a911a-4071-11e8-89f7-0136a3c47c70.png)    
这个和 Yimu 之前分享的[ Age Progression/Regression by Conditional Adversarial Autoencoder ](https://arxiv.org/abs/1702.08423)类似

---

#### 作者还将 Image Synthesis GAN 粗略归为3类 direct methods,iterative methods and hierarchical methods：
![2018 04 15 3methods](https://user-images.githubusercontent.com/10448025/38772804-b21d47b6-4071-11e8-8349-3e3f5d0af347.png)    

---
#### 在 Section.7 作者还描绘了一个炼丹自动化的趋势，感觉很有意思

> Traditional pattern recognition and machine learning require us to define what features to be used (e.g. SIFT and HOG), and we design specific loss functions and decide what optimization methods to be applied. 

> Deep learning free us from carefully designing features, by learning low-level and high-level feature representations by itself during training , but we still need to work hard at designing loss functions that work well. 

> GAN takes us one step forward.Discriminator acts as a **learned loss function**, that it learns how to evaluate data samples instead of being told how to do so,  the discriminator learns how to calculate how well an output matches the true data distribution. Such property allows models to be more flexible and more likely to generalize well. 

>Further-more, with `learn2learn` which allows neural networks to learn to optimize themselves, there is a possibility that we may no longer need to choose what optimizers (such as RMSprop, Adam) to use and let models handle everything themselves

## Next

 * 继续刷题

 * 跑一个人脸生成的实验

 * 重构整改[`yllab`](https://github.com/DIYer22/yllab) ，并兼容 Python 3
